bcc/module.go

```diff
@@ -201,6 +229,74 @@ func (bpf *Module) AttachKretprobe(fnName string, fd int) error {
 	return bpf.attachProbe(evName, BPF_PROBE_RETURN, fnName, fd)
 }
 
+// AttachUprobe attaches a uprobe fd to the symbol in the library or binary 'name'
+// The 'name' argument can be given as either a full library path (/usr/lib/..),
+// a library without the lib prefix, or as a binary with full path (/bin/bash)
+// A pid can be given to attach to, or -1 to attach to all processes
+func (bpf *Module) AttachUprobe(name, symbol string, fd, pid int) error {
+	path, addr, err := resolveSymbolPath(name, symbol, 0x0, 0)
```

Comment:
Does it work when attaching a uprobe to a process in a container?

This causes a call to `C.bcc_resolve_symname()` with  the parameter `pid=0`. bcc temporarily changes the current mount namespace to the mount namespace of the specified pid (if specified, i.e. !=0) in order to read the symbols from the correct files in the correct rootfs when reading `/proc/<pid>/maps`.

Since the pid is not specified here, I suspect `AttachUprobe` does not work correctly when the traced process lives in a different mount namespace than the tracer (like with containers, or with daemons started by systemd with [RootDirectory=](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#RootDirectory=)).

Unfortunately, I think just passing the pid in parameter would not work because "man 2 setns" says:
> A process may not be reassociated with a new mount namespace if it is multithreaded.

And Go programs are always multithreaded.

rkt works around a similar problem by [forking a new single-threaded process to do the setns](https://github.com/rkt/rkt/blob/cb25d9062afdc7ca4fe7d1ebd2c106d2fcee36ed/stage1/app_add/app_add.go#L223-L226).

If this is correct, I suggest to add a comment about this or file a follow-up issue..